lib/cocoapods/sandbox/headers_store.rb

```diff
@@ -86,23 +86,17 @@ def add_files(namespace, relative_header_paths)
       #         the path of the header file relative to the Pods project
       #         (`PODS_ROOT` variable of the xcconfigs).
       #
-      # @param  [String] final_name
-      #         the name under which the file should be available in the
-      #         headers directory.
-      #
       # @note   This method does _not_ add the file to the search paths.
       #
       # @return [Pathname]
       #
-      def add_file(namespace, relative_header_path, final_name)
+      def add_file(namespace, relative_header_path)
         namespaced_path = root + namespace
         namespaced_path.mkpath unless File.exist?(namespaced_path)
 
         absolute_source = (sandbox.root + relative_header_path)
         source = absolute_source.relative_path_from(namespaced_path)
-        Dir.chdir(namespaced_path) do
-          FileUtils.ln_sf(source, final_name)
-        end
+        FileUtils.ln_sf(source, namespaced_path)
```

Comment:
Shouldn't this be `FileUtils.ln_sf(source, File.join(namespaced_path, relative_header_path.basename))`? It looks like the final arg is now a directory path instead of a file path, and linking a header to a directory would be a change in behavior.